pkg/apiserver/resthandler_test.go

```diff
@@ -149,12 +150,12 @@ type patchTestCase struct {
 	admit updateAdmissionFunc
 
 	// startingPod is used as the starting point for the first Update
-	startingPod *api.Pod
+	startingPod *v1.Pod
```

Comment:
not sure what the point of switching all of these to v1.Pod is... the runtime.Encode was already ensuring we had v1 JSON as our patches. What the codec does (even when encoding a v1.Pod to JSON) is opaque enough that if you want to really test something like the deprecated service account field, I'd suggest setting up a separate test that actually uses JSON strings instead of relying on a codec step inside the test loop